core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java

```diff
@@ -57,19 +56,86 @@ void setProperty(final String property, final String value)
+   *               //The endsWith() function was used to make the example short, would be better to use equals().
+   *               if(planner != null && planner.endsWith("DefaultCompactionPlanner") {
+   *                 // tservers will eventually see these compaction planner changes and when they do they will see all of the changes at once
+   *                 currProps.keySet().removeIf(
+   *                    prop -> prop.startsWith("tserver.compaction.major.service.default.planner.opts."));
+   *                 currProps.put("tserver.compaction.major.service.default.planner","MyPlannerClassName");
+   *                 currProps.put("tserver.compaction.major.service.default.planner.opts.myOpt1","val1");
+   *                 currProps.put("tserver.compaction.major.service.default.planner.opts.myOpt2","val2");
+   *                }
+   *             });
+   *
+   *             // Since three properties were set may want to check for the values of all
+   *             // three, just checking one in this example to keep it short.
+   *             if("MyPlannerClassName".equals(acceptedProps.get("tserver.compaction.major.service.default.planner"))){
+   *                // the compaction planner change was accepted or already existed, so take action for that outcome
+   *             } else {
+   *                // the compaction planner change was not done, so take action for that outcome
+   *             }
+   *           }
+   *         }
+   * </pre>
+   *
+   * @param mapMutator
+   *          This consumer should modify the passed snapshot of instance properties to contain the
+   *          desired keys and values. It should be safe for Accumulo to call this consumer multiple
+   *          times, this may be done automatically when certain retryable errors happen. The
+   *          consumer should probably avoid accessing the Accumulo client as that could lead to
+   *          undefined behavior.
+   *
    * @throws AccumuloException
    *           if a general error occurs
    * @throws AccumuloSecurityException
    *           if the user does not have permission
    * @throws IllegalArgumentException
    *           if the Consumer alters the map by adding properties that cannot be stored in
    *           ZooKeeper
-   * @throws ConcurrentModificationException
-   *           without altering the stored properties if the server reports that the properties have
-   *           been changed by another process
+   *
+   * @return The map that became Accumulo's new properties for this table. This map is immutable and
+   *         contains the snapshot passed to mapMutator and the changes made by mapMutator.
+   * @since 2.1.0
    */
-  void modifyProperties(Consumer<Map<String,String>> mapMutator) throws AccumuloException,
-      AccumuloSecurityException, IllegalArgumentException, ConcurrentModificationException;
+  Map<String,String> modifyProperties(Consumer<Map<String,String>> mapMutator)
```

Comment:
I think `computeProperties` might also be a decent name, since it is now returning the properties. This is analogous to the `compute*()` methods that return the resulting value in the Java Map APIs.

Reply:
I think i like `modify` a bit better because it conveys something that is happening in the distributed system where `compute` feels a bit more centered on what is happening in the client.